src/core/thread/link_metrics.cpp

```diff
@@ -809,7 +809,8 @@ int8_t ScaleRawValueToRssi(uint8_t aRawValue)
     int32_t value = aRawValue;
 
     value = value * (kMaxRssi - kMinRssi);
-    value = DivideAndRoundToClosest<int32_t>(value, NumericLimits<uint8_t>::kMax);
+    // 0 or 1 plus kMinRssi (-130) will cause overflow on int8_t
+    value = Max(static_cast<int32_t>(2), DivideAndRoundToClosest<int32_t>(value, NumericLimits<uint8_t>::kMax));
     value += kMinRssi;
 
     return static_cast<int8_t>(value);
```

Comment:
Good find!
How about we clamp it here when we cast (limit it to `int8` min value)?
```suggestion
    return static_cast<int8_t>(Max(value, static_cast<int32>(NumericLimits<int8_t>::kMin));
```

Reply:
Or (maybe even better) we can define  `ClampToInt8()` in `num_utils.hpp` (similar to `ClampToUint8()`) and then use it here?

Reply:
I added the clamp method. But `Int` can be clamped on two sides: left and right. In most cases we only need to clamp it on one side so I think we don't want to clamp on both sides in one method. I named the new method as `ClampToInt8L` (L for Left).